### PR TITLE
#34 Add code coverage support

### LINKED ISSUE #34: Add code coverage support
## Background
Add code coverage support to be able measure code quality.

## Feature
New ability to measure current code coverage as one of QA metrics.

### PR DESCRIPTION
**Pull request by miroslavpojer**
- new plugin dependency on jacoco library
- how to section in `readme` file.

Closes #34

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,3 +53,12 @@ Spark and Scala compatibility
 ## How to Release
 
 Please see [this file](RELEASE.md) for more details.
+
+## How to generate Code coverage report
+```sbt
+sbt jacoco
+```
+Code coverage will be generated on path:
+```aidl
```
Comment (Zejnilovic) [inline README.md:62]: `aidl`? Android? Or is it something else?

**Review by cerveada — APPROVED** [review]
LGTM

**Review by dk1844 — APPROVED** [review]
- [x] code reviewed
- [x] pulled
- [x] built
- [x] run and briefly checked out the generated html

Nice, tried, seems to work.

File: README.md (hunk 1 above)
Comment (miroslavpojer) [inline README.md:62]:
```suggestion
```
```

Comment (miroslavpojer) [reply README.md]: Wrong tag. Removed in

**Review by dk1844 — APPROVED** [review]

**Review by lsulak — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -53,3 +53,12 @@ Spark and Scala compatibility
...
 
 Please see [this file](RELEASE.md) for more details.
+
+## How to generate Code coverage report
+```sbt
+sbt jacoco
+```
+Code coverage will be generated on path:
+```
+{local-path}\spark-data-standardization\target\scala-2.XY\jacoco\report\html
```
Comment (Zejnilovic) [inline README.md:63]:
```suggestion
{project-root}/target/scala-2.XY/jacoco/report/html
```

File: build.sbt [hunk 3]
```diff
@@ -57,3 +57,17 @@ scalafmtFilter.withRank(KeyRanks.Invisible) := fmtFilterExpression
...
 // linting
 Global / excludeLintKeys += ThisBuild / name // will be used in publish, todo #3 - confirm if lint ignore is still needed
+
+// JaCoCo code coverage
+Test / jacocoReportSettings := JacocoReportSettings(
+  s"spark-data-standardization Jacoco Report - ${scalaVersion.value}",
+  None,
+  JacocoThresholds(),
+  Seq(JacocoReportFormats.HTML, JacocoReportFormats.XML),
+  "utf-8")
```
Comment (Zejnilovic) [inline build.sbt:67]:
```suggestion
  "utf-8"
)
```

File: build.sbt [hunk 4]
```diff
@@ -57,3 +57,17 @@ scalafmtFilter.withRank(KeyRanks.Invisible) := fmtFilterExpression
...
+
+// exclude example
+//Test / jacocoExcludes := Seq(
+//  "za.co.absa.standardization.udf.UDFBuilder*", // class and related objects
+//  "za.co.absa.standardization.udf.UDFNames" // class only
+//)
```
Comment (Zejnilovic) [inline build.sbt:73]: I don't think this should be here. I would delete this as it does not add value, and it should be in Jacoco documentation.

Comment (miroslavpojer) [reply build.sbt:74]: I have prepared this example in each touched project as preparation for next phase. I can delete it, sure.

Comment (benedeki) [reply build.sbt:74]: I actually liked it. 🤷  No need to search docu.

**Review by dk1844 — APPROVED** [review]

**Pull request merged**